homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,234 @@
+"""Config flow for Flux LED/MagicLight."""
+import copy
+import logging
+
+from flux_led import BulbScanner
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    CONF_AUTOMATIC_ADD,
+    CONF_CONFIGURE_DEVICE,
+    CONF_DEVICES,
+    CONF_EFFECT_SPEED,
+    CONF_REMOVE_DEVICE,
+    DEFAULT_EFFECT_SPEED,
+    DOMAIN,
+    SIGNAL_ADD_DEVICE,
+    SIGNAL_REMOVE_DEVICE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for FluxLED/MagicHome Integration."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry):
+        """Get the options flow for the Flux LED component."""
+        return OptionsFlow(config_entry)
+
+    async def async_step_import(self, data: dict = {}):
```

Comment:
Ideally we create a separate config entry for each host.  This avoids the need to build retry logic to reconnect to a device if it's offline during startup as we can raise ConfigEntryNotReady and auto retry later

Reply:
My concern is that the only way to fire the scanner currently is through broadcasting.  It doesn't target specific IP's yet (future update).  I'll double-check on if it reports the MAC/Model directly as those would be ideal to use as the IDs.

Reply:
If you send the packet directly to the ip, does it respond?

Reply:
```python
diff --git a/flux_led/__main__.py b/flux_led/__main__.py
index 98c6488..cb4c2d8 100755
--- a/flux_led/__main__.py
+++ b/flux_led/__main__.py
@@ -1291,7 +1291,7 @@ class BulbScanner:
             if time.time() > quit_time:
                 break
             # send out a broadcast query
-            sock.sendto(msg, ("<broadcast>", DISCOVERY_PORT))
+            sock.sendto(msg, ("192.168.214.166", DISCOVERY_PORT))
 
             # inner loop waiting for responses
             while True:
```

```
bdraco@travelmacjnk-4 flux_led % python3 -m flux_led -sSti                   
18B905C5239B [192.168.214.166] ON  [Color: (0, 255, 0) White: 0 raw state: 129,51,35,97,35,1,0,255,0,0,9,0,240,84,]
  Timer #1: Unset
  Timer #2: Unset
  Timer #3: Unset
  Timer #4: Unset
  Timer #5: Unset
  Timer #6: Unset
```

Looks like they do respond

Reply:
> If you send the packet directly to the ip, does it respond?

Yes, they respond.  I just need to update flux_led to make it work and didn't have time. I was going to update it after this PR is finished.